src/main/java/Duke/Parser.java

```diff
@@ -0,0 +1,88 @@
+package Duke;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Scanner;
+
+/**
+ * The Parser class is responsible for parsing user input and generating corresponding commands.
+ */
+public class Parser {
+    private static final Scanner scanner = new Scanner(System.in);
+
+    /**
+     * Gets user input from the console.
+     *
+     * @return The user input as a string.
+     */
+    public static String getUserInput() {
+        return scanner.nextLine();
+    }
+
+    /**
+     * Parses the task index from the user input.
+     *
+     * @param userInput The user input containing the task index.
+     * @return The parsed task index.
+     * @throws DukeException If an error occurs while parsing the task index.
+     */
+    public static int parseTaskIndex(String userInput) throws DukeException {
+        try {
+            // Assuming the input is in the format "mark 2" or "unmark 2"
+            String indexString = userInput.split(" ", 2)[1].trim();
+            return Integer.parseInt(indexString) - 1;
+        } catch (NumberFormatException | ArrayIndexOutOfBoundsException e) {
+            throw new DukeException("Please enter a valid task index :(");
+        }
+    }
+
+    /**
+     * Checks if the user input is an exit command.
+     *
+     * @param userInput The user input to check.
+     * @return True if the input is an exit command, false otherwise.
+     */
+    public static boolean isExitCommand(String userInput) {
```

Comment:
boolean variables should be in the form of is...however this is a method so maybe it should still be a verb? Such as "checkIfExitCommand"